components/ChipTextField/examples/MDCChipTextField.m

```diff
@@ -0,0 +1,360 @@
+                                   relatedBy:NSLayoutRelationEqual
+                                      toItem:self
+                                   attribute:NSLayoutAttributeTop
+                                  multiplier:1
+                                    constant:0];
+  NSLayoutConstraint *chipContainerViewConstraintBottom =
+      [NSLayoutConstraint constraintWithItem:self.chipsContainerView
+                                   attribute:NSLayoutAttributeBottom
+                                   relatedBy:NSLayoutRelationEqual
+                                      toItem:self
+                                   attribute:NSLayoutAttributeBottom
+                                  multiplier:1
+                                    constant:0];
+  [NSLayoutConstraint
+      activateConstraints:@[ chipContainerViewConstraintTop, chipContainerViewConstraintBottom ]];
+  [self updateChipViewLeadingConstraints];
+  [self updateChipViewTrailingConstraints];
+}
+
+- (void)appendChipWithText:(NSString *)text {
+  MDCChipView *chipView = [[MDCChipView alloc] init];
+  chipView.titleLabel.text = text;
+  chipView.translatesAutoresizingMaskIntoConstraints = NO;
+
+  [self.chipsContainerView appendChipView:chipView];
+
+  // recalculate the layout to get a correct chip frame values
+  [self.chipsContainerView layoutIfNeeded];
+  [self.chipViews addObject:chipView];
+
+  [self clearChipsContainerOffsetWithConstant:kTextToEnterPlaceholderLength];
+}
+
+// TODO: the constant here reflects the margin, places calling this method need to be refactored.
+- (void)clearChipsContainerOffsetWithConstant:(CGFloat)constant {
+  self.chipContainerViewConstraintTrailingConstant = -constant;
+  [self invalidateIntrinsicContentSize];
+  [self setNeedsLayout];
+  [self layoutIfNeeded];
+  [self.chipsContainerView setNeedsLayout];
+  [self.chipsContainerView layoutIfNeeded];
+  [self.chipsContainerView scrollToRight];
+}
+
+#pragma mark - Text Editing Handler
+
+- (void)chipTextFieldTextDidChange {
+  [self deselectAllChips];
+  [self setupEditingRect];
+}
+
+- (void)setupEditingRect {
+  if (CGRectGetWidth(self.chipsContainerView.frame) <= 0) {
+    return;
+  }
+
+  UITextRange *textRange = [self textRangeFromPosition:self.beginningOfDocument
+                                            toPosition:self.endOfDocument];
+  CGRect inputRect = [self firstRectForRange:textRange];
+  CGFloat inputRectLength = inputRect.size.width;
+
+  self.chipContainerViewConstraintTrailingConstant =
+      -(inputRectLength + kTextToEnterPlaceholderLength);
+  [self setNeedsUpdateConstraints];
+  [self setNeedsLayout];
+  [self layoutIfNeeded];
+  [self.chipsContainerView setNeedsLayout];
+  [self.chipsContainerView layoutIfNeeded];
+  [self.chipsContainerView scrollToRight];
+}
+
+#pragma mark - Constraints
+
+- (void)updateConstraints {
+  // TODO: This is not optimized for performance, but due to how MDCTextInputController works, we
```

Comment:
Not clear what's the "todo" (the action that needs to be taken) here. Can you explain? Also, please add to our "todo" list if there's one.

Reply:
Good point. Addressed in327c22b99d72535ea92b932b3286dbf257e542bb. No new constraints are created in this method now.